server/server.ts

```diff
@@ -315,6 +323,83 @@ app.get('/api/fetchlikes/:userId', authMiddleware, async (req, res, next) => {
   }
 });
 
+app.post('/api/sendemail', async (req, res, next) => {
+  const reqBody = req.body;
+  const objectData = reqBody.objectData as EmailSentForm;
+  const toAddress = objectData.toAddress;
+  const fromAddress = objectData.fromAddress;
+  const emailTitle = objectData.title;
+  const message = objectData.message;
+
+  if (!objectData) {
+    throw new ClientError(400, 'email form object can not be undefined.');
+  }
+  const recipe: Recipe = reqBody.recipe;
+  if (!recipe) {
+    res.json('recipe undefined');
+    return;
+  }
+
+  const createSendEmailCommand = (
+    toAddress: string,
+    fromAddress: string,
+    subject: string,
+    recipe: Recipe,
+    message: string
+  ): any => {
+    return new SendEmailCommand({
+      Destination: {
+        CcAddresses: [],
+        ToAddresses: [toAddress],
+      },
+      Message: {
+        Body: {
+          Html: {
+            Charset: 'UTF-8',
+            Data: `<div><div>${message}</div><br/><div>${recipe.ingredients}</div><br/>
+            <div>${recipe.instructions}</div></div>`,
+          },
+          Text: {
+            Charset: 'UTF-8',
+            Data: '',
+          },
+        },
+        Subject: {
+          Charset: 'UTF-8',
+          Data: subject,
+        },
+      },
+      Source: fromAddress,
+      ReplyToAddresses: [],
+    });
+  };
+
+  const run = async (): Promise<any> => {
+    const sendEmailCommand = createSendEmailCommand(
+      toAddress,
+      fromAddress,
+      emailTitle,
+      recipe,
+      message
+    );
+    try {
+      const dataSent = await sesClient.send(sendEmailCommand);
+      return dataSent;
+    } catch (e) {
+      console.error(e);
+      return e;
+    }
+  };
+
+  try {
+    run();
```

Comment:
This is unusual code. I think it would read more naturally if you just put the code here, instead of creating another function and then calling that function here (two functions, actually). It would look something like:

```
try {
  const sendEmailCommand = new SendEmailCommand({
    Destination: ....
  });
  return await sesClient.send(sendEmailCommand);
} catch (e) {
 ...
}
```

Or if you still wanted a separate `createSendEmailCommand` function, it would look like:

```
try {
  const sendEmailCommand = createSendEmailCommand(toAddress, fromAddress, etc);
  return await sesClient.send(sendEmailCommand);
} catch (e) {
 ...
}
```